torch/csrc/jit/register_special_ops.cpp

```diff
@@ -150,6 +150,68 @@ void recursiveStore(
   }
 }
 
+template<bool if_set_requires_grad>
+Operation createTensorFromList(const Node* node) {
+  auto input = node->inputs().at(0);
+  auto elem_type = input->type();
+  while (auto list_type = elem_type->cast<ListType>()) {
+    elem_type = list_type->getElementType();
+  }
+  checkListInputType(elem_type, node);
+  at::ScalarType initial_scalar_type = scalarTypeFromJitType(elem_type);
+  return [initial_scalar_type, elem_type](Stack& stack) {
+    bool requires_grad;
+    IValue data;
+    IValue dtype;
+    IValue device;
+    if (if_set_requires_grad) {
+      pop(stack, data, dtype, device, requires_grad);
```

Comment:
it is kind of hard to follow that `requires_grad` is only conditionally set if `if_set_requires_grad` a templated parameter is set to true.

Also In order to check if this function is correct you have to look back at all of its callsites and see if if the schema matches to verify correctness, which kind of breaks abstraction of the function.

Reply:
what do you mean by this? this abstraction only differed in a flag where one schema contains requires_grad and the other not

Reply:
maybe add comment that torch::tensor has a fourth `requires_grad` arg that `as_tensor` does not have